encodings/fastlanes/src/bitpacking/compute/filter.rs

```diff
@@ -13,6 +13,11 @@ use crate::{BitPackedArray, BitPackedEncoding};
 
 impl FilterFn<BitPackedArray> for BitPackedEncoding {
     fn filter(&self, array: &BitPackedArray, mask: FilterMask) -> VortexResult<ArrayData> {
+        // If the indices are large enough, it's faster to flatten and take the primitive array.
+        if mask.true_count() * UNPACK_CHUNK_THRESHOLD > array.len() {
+            return filter(&array.clone().into_canonical()?.into_array(), mask);
```

Comment:
This is going to mean if more than 12.5% of the mask is true, it will canonicalize/allocate the whole thing. I'd expect we can do better by just iterating 1024 elements at a time, unpacking to an array on the stack (like take does)?

Reply:
That's correct. Whether that's what we want, I'm not sure. If it is, I imagine it's probably not the same threshold as unpack_single.

Collecting indices for each chunk is actually relatively expensive, vs streaming through all chunks. We should try to optimise the chunked iterator I think.